gpt_index/tts/base.py

```diff
@@ -0,0 +1,77 @@
+"""Text to speech module."""
+
+from typing import Optional
+import tempfile
+import os
+import numpy as np
+
+
+class BaseTTS:
+    def __init__(self) -> None:
+        pass
+
+
+class BarkTTS(BaseTTS):
+    def __init__(
+        self,
+        text_temp: float = 0.7,
+        waveform_temp: float = 0.7,
+        lang_speaker_voice: Optional[str] = None,
+    ) -> None:
+        """
+        Args:
+            text_temp: generation temperature (1.0 more diverse, \
+                0.0 more conservative)
+            waveform_temp: generation temperature (1.0 more diverse, \
+                0.0 more conservative)
+            lang_speaker_voice: language speaker voice for audio cloning.
+        """
+
+        super().__init__()
+
+        import_err_msg = "`bark` package not found, \
+            please run `pip install git+https://github.com/suno-ai/bark.git`"
+        try:
+            import bark
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        self.generate_audio_fn = bark.generate_audio
```

Comment:
nit: could you use e.g. `self._generate_audio_fn` to mark these attributes as private?

Reply:
Unless you have a strong reason, it's better to reveal as little as possible about the underlying implementation of an object.

Reply:
Also, why do we need to save bark.generate_audio as an instance variable?

Reply:
if not saved as an instance variable I need to `import bark` in the `generate_bark_audio` function as well. So having it as an instance variable makes sense here?